spec/webauthn/authenticator_assertion_response_spec.rb

```diff
@@ -7,69 +7,98 @@
   let(:encoded_challenge) { WebAuthn::Utils.ua_encode(challenge) }
   let(:original_origin) { fake_origin }
 
-  it "is valid if everything's in place" do
-    assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-      client_data_json: encoded_fake_client_data_json(challenge: challenge,
-                                                      origin: original_origin,
-                                                      type: "webauthn.get"),
-      authenticator_data: fake_authenticator_data
+  let(:client_data_json) { fake_client_data_json(challenge: challenge, origin: original_origin, type: "webauthn.get") }
+  let(:encoded_client_data_json) { WebAuthn::Utils.ua_encode(client_data_json) }
+  let(:credential_key) { fake_credential_key }
+  let(:authenticator_data) { fake_authenticator_data(credential_public_key: credential_key.public_key) }
+
+  let(:signature) do
+    fake_signature(
+      key: credential_key,
+      authenticator_data: authenticator_data,
+      client_data_json: client_data_json
     )
+  end
 
-    expect(assertion_response.valid?(encoded_challenge, original_origin)).to be_truthy
+  let(:assertion_response) do
+    WebAuthn::AuthenticatorAssertionResponse.new(
+      client_data_json: encoded_client_data_json,
+      authenticator_data: authenticator_data,
+      signature: signature
+    )
   end
 
-  describe "type validation" do
-    it "is invalid if type is not get" do
-      assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-        client_data_json: encoded_fake_client_data_json(challenge: challenge,
-                                                        origin: original_origin,
-                                                        type: "webauthn.create"),
-        authenticator_data: fake_authenticator_data
+  it "is valid if everything's in place" do
+    expect(
+      assertion_response.valid?(
+        encoded_challenge,
+        original_origin,
+        credential_public_key: key_bytes(credential_key.public_key)
+      )
+    ).to be_truthy
+  end
+
+  it "is invalid if signature was signed with a different key" do
+    different_key = fake_credential_key
+
+    expect(
+      assertion_response.valid?(
+        encoded_challenge,
+        original_origin,
+        credential_public_key: key_bytes(different_key.public_key)
       )
+    ).to be_falsy
+  end
 
-      expect(assertion_response.valid?(encoded_challenge, original_origin)).to be_falsy
+  describe "type validation" do
+    let(:client_data_json) { fake_client_data_json(challenge: challenge, origin: original_origin, type: "webauthn.create") }
+
+    it "is invalid if type is not get" do
+      expect(
+        is_valid = assertion_response.valid?(
```

Comment:
I think this variable is not necessary. Maybe a leftover from a diff implementation idea.

Reply:
Good catch. Yeah, leftovers.